### PR TITLE
feat: tabs support cssVar

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | tabs support css variable theme |
| 🇨🇳 Chinese | 标签页支持 css 变量主题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 0fd6677</samp>

This pull request enhances the tabs component with CSS variables support and refactors its style code. It adds custom hooks and a provider component to enable dynamic theming and customization of the tabs component and its dropdown menu. It also improves the consistency and compatibility of the CSS variables and calculations by using the `@ant-design/cssinjs` package and extracting common functions to separate files.

### <samp>🤖 Generated by Copilot at 0fd6677</samp>

*  Import and use custom hooks for registering and applying CSS variables for the tabs component ([link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-29d06a1647b854ac9e4eff3f5e9f5b25dca5e02a92ddde121a1af24940c767adR17-R18), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-29d06a1647b854ac9e4eff3f5e9f5b25dca5e02a92ddde121a1af24940c767adL56-R62), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-29d06a1647b854ac9e4eff3f5e9f5b25dca5e02a92ddde121a1af24940c767adL89-R94), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-29d06a1647b854ac9e4eff3f5e9f5b25dca5e02a92ddde121a1af24940c767adL107-R114), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-0d8e0ae6ba41e2ef504286c8ca0332f966027ec81ad2b75a3e196ae922b0f52eR1-R4), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7R1014-R1049))
*  Wrap numeric token values with the `unit` function to convert them to CSS units ([link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L157-R163), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L177-R183), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L186-R192), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L198-R204), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L211-R217), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L221-R227), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L239-R245), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L277-R283), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L297-R303), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L371-R377), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L433-R439), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L454-R460), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L530-R540), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L566-R572), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L618-R629), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L630-R636), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L638-R644), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L780-R786), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L787-R797), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L941-R950))
*  Replace some token values with the `token.calc` function to create CSS calculation expressions ([link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L454-R460), [link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L929-R935))
*  Add a new token property `tabsCardLineHeight` to the `ComponentToken` interface ([link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7R129-R133))
*  Replace the `token.cardPadding` value with the `token.tabsCardPadding` value, which is calculated from the `token.cardPadding` value if it is empty ([link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L1014-R1062))
*  Update the imports in the `style/index.ts` file, replacing the `FullToken` type with the `GetDefaultToken` type and removing the `genComponentStyleHook` import ([link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L1-R4))
*  Simplify the `genComponentStyleHook` function call by using the `prepareComponentToken` function instead of a function expression ([link](https://github.com/ant-design/ant-design/pull/45803/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7L1037-R1075))

### CODE REVIEW

File: components/tabs/index.tsx [hunk 1]
```diff
@@ -104,8 +109,9 @@ const Tabs: React.FC<TabsProps> & { TabPane: typeof TabPane } = (props) => {
         className,
         rootClassName,
         hashId,
+        rootCls,
       )}
-      popupClassName={classNames(popupClassName, hashId)}
+      popupClassName={classNames(popupClassName, hashId, rootCls)}
```
Comment (MadCcc) [inline components/tabs/index.tsx:114]: 这里没有 rootClassName 么

Comment (kiner-tang) [reply components/tabs/index.tsx:114]: 以前是没有的

File: components/tabs/style/index.ts [hunk 2]
```diff
@@ -430,7 +436,7 @@ const genPositionStyle: GenerateStyle<TabsToken> = (token: TabsToken): CSSObject
     [`${componentCls}-bottom`]: {
       [`> ${componentCls}-nav, > div > ${componentCls}-nav`]: {
         order: 1,
-        marginTop: `${margin}px`,
+        marginTop: `${unit(margin)}`,
```
Comment (MadCcc) [inline components/tabs/style/index.ts:439]:
```suggestion
        marginTop: margin,
```

File: components/tabs/style/index.ts [hunk 3]
```diff
@@ -125,6 +126,11 @@ export interface ComponentToken {
    * @descEN Gutter of card tab
    */
   cardGutter: number;
+  /**
+   * @desc 卡片标签的行高
+   * @descEN height of card tab
+   */
+  tabsCardLineHeight: number | string;
```
Comment (MadCcc) [inline components/tabs/style/index.ts:133]:
```suggestion
  cardLineHeight: number | string;
```

File: components/tabs/style/index.ts [hunk 4]
```diff
@@ -1005,23 +1011,55 @@ const genTabsStyle: GenerateStyle<TabsToken> = (token: TabsToken): CSSObject =>
...
+    horizontalItemPaddingSM: `${token.paddingXS}px 0`,
+    horizontalItemPaddingLG: `${token.padding}px 0`,
+    verticalItemPadding: `${token.paddingXS}px ${token.paddingLG}px`,
+    verticalItemMargin: `${token.margin}px 0 0 0`,
+    itemColor: token.colorText,
+    itemSelectedColor: token.colorPrimary,
+    itemHoverColor: token.colorPrimaryHover,
+    itemActiveColor: token.colorPrimaryActive,
+    cardGutter: token.marginXXS / 2,
+    tabsCardLineHeight: Math.round(token.fontSize * token.lineHeight),
```
Comment (MadCcc) [inline components/tabs/style/index.ts:1046]: 用 fontHeight

**Review by MadCcc — APPROVED** [review]

**Pull request merged**